enzy_htp/structure/structure_region/residue_caps.py

```diff
@@ -0,0 +1,579 @@
+    def chain(self) -> Chain:
+        """Gets the chain name of the link_residue."""
+        return self.link_residue.parent
+
+    @property
+    def name(self) -> str:
+        """Gets the residue name of the link_residue."""
+        return self.link_residue.name
+    
+    @property
+    def term_type(self) -> str:
+        """What type of capping is this? n-term, c-term, etc."""
+        return self._terminal_type
+
+    @term_type.setter
+    def term_type(self, val_in:str) -> None:
+        """Setter for the terminal type of the ResidueCap(). Does not perform validation."""
+        self._terminal_type = val_in
+
+    def init_fixed_atomic_charges(self) -> None:
+        """init hard coded fixed atomic charges.
+        Used when init_charge and fixed charge stragety is used."""
+        for atom in self.atoms:
+            atom.charge = self.FIXED_CHARGE_MAP[atom.name]
+
+    def is_residue_cap(self) -> bool:
+        """Does this class cap a Residue? Always True."""
+        return True
+
+    def is_nterm_cap(self) -> bool:
+        """Does this ResidueCap attach to the n-terminal end of an amino acid Residue?"""
+        return self.term_type == "nterm"
+
+    def is_cterm_cap(self) -> bool:
+        """Does this ResidueCap attach to the c-terminal end of an amino acid Residue?"""
+        return self.term_type == "cterm"
+
+    def __str__(self) -> str:
+        """str() rep of the ResidueCap() that shows its chemical type as well as what Atom()'s it is linked to."""
+        return f"{self.cap_type}Cap(link_atom={self.link_atom.name},link_residue={self.link_residue.key_str})"
+
+
+class HCap(ResidueCap):
+    """A ResidueCap() which uses a -H (hydrogen) as a cap. Adds a total of 1 atom1, with the 
+    
+    n-terminal version having atom names:       HP11 
+    and c-terminal version having atom names:   HP21
+
+    Can generally be used for both n-terminal/c-teriminal and other non-amino acid bonds
+    """
+    FIXED_CHARGE_MAP = {
+        # C-ter
+        "HP21" : 0.0,
+        # N-ter
+        "HP11" : 0.0,
+    } # improve this
+    """Fixed charge for a -H attached to the N/C-ter
+    of a residue."""
+
+    CAP_BOND_DISTANCE:Dict[str, float] = {
+        "N" : 1.1,
+        "C" : 1.0,
+    }
+    """Maps the name of the link_atom to appropriate bond distance."""
+
+    @property
+    def cap_type(self) -> str:
+        """hard coded cap type"""
+        return "H"
+
+    @property
+    def net_charge(self) -> str:
+        """hard coded net charge"""
+        return 0
+
+    def anchor(self) -> None:
+        """Straightforward anchoring that is identical for both n-terminal and c-terminal capping."""
+        p1 = np.array(self.link_atom.coord)
+        p2 = np.array(self.socket_atom.coord)
+
+        direction = p2 - p1
+        direction /= np.linalg.norm(direction)
+    
+        cap_bond_distance = self.CAP_BOND_DISTANCE[self.link_atom.name]
+
+        self.atoms[0].coord = p1 + cap_bond_distance*direction
```

Comment:
Be great if we can put these math in math_helper.

Reply:
I refined this as much as possible and put a simpler implementation in!

Reply:
Basically I put a method in `StructureRegion` called `align_rigid_atoms()` which is called in all `ResidueCap` children except for `HCap` (for obvious reasons). I could not (or at least did not want to) put anything in `core` since the point is to operate on `Atom()` objects and obviously there are not `Atom()'s in `core`. I did move a function called `direction_unit_vector()` to `math_helper.py` since I was using that function a lot. There are like three other comments where I just reference this bigger comment, but the same approach is used elsewhere.

Reply:
Cool!
But I think for these
```
            pt = np.transpose(np.matmul(rot_mat, np.transpose( pt  )))
            pt += (d0*bd) + p0
```
they can still go to core and be like apply rotation_matrix() or I bet you can find a function from `from scipy.spatial.transform import Rotation`  to do that

Reply:
Good point! We will create a more permanent solution at some point in the future. See issue https://github.com/ChemBioHTP/EnzyHTP/issues/164

Reply:
For reference I have tried to use `scipy.spatial.transform.Rotation` in the past but it was generally not very easy to use and I struggled to find a way to implement it especially given that our needs are not very robust.